Kernel/FileSystem/Ext2Base.h

```diff
@@ -0,0 +1,307 @@
+                 min_fragment_log_size = min_block_log_size;
+
+// The size of an inode
+constexpr size_t good_old_inode_size = 128;
+
+// This is how many hard links are allowed at maximum.
+constexpr u32 link_max = 65000;
+
+enum class FeatureFullCompat : u32 {
+    DirPrealloc = (1 << 0),
+    ImagicInodes = (1 << 1),
+    HasJournal = (1 << 2),
+    ExtendedAttrs = (1 << 3),
+    ResizeInode = (1 << 4),
+    DirIndex = (1 << 5),
+};
+
+enum class FeatureReadOnlyCompat {
+    SparseSuper = (1 << 0),
+    LargeFile = (1 << 1),
+};
+
+AK_ENUM_BITWISE_OPERATORS(FeatureFullCompat);
+AK_ENUM_BITWISE_OPERATORS(FeatureReadOnlyCompat);
+
+// Special inode numbers
+enum class InodeNums : u64 {
+    Bad = 1,     // Bad blocks inode
+    Root,        // Root inode
+    ACLIndex,    // ACL inode
+    ACLData,     // ACL inode
+    BootLoader,  // Boot loader inode
+    UndeleteDir, // Undelete directory inode
+    Resize,      // Reserved group descriptors inode
+    Journal,     // Journal inode
+
+    FirstInode = 11, // First usable inode
+};
+
+// File system states
+enum class FileSystemState : u32 {
+    Valid,  // Cleanly unmounted.
+    Error,  // Errors detected
+    Orphan, // EXT3 Orphans being recovered
+};
+
+// Structure of a block group descriptor
+struct GroupDescriptor {
+    u32 bg_block_bitmap;      // Blocks bitmap block
+    u32 bg_inode_bitmap;      // Inodes bitmap block
+    u32 bg_inode_table;       // Inodes table block
+    u16 bg_free_blocks_count; // Free blocks count
+    u16 bg_free_inodes_count; // Free inodes count
+    u16 bg_used_dirs_count;   // Directories count
+    u16 bg_flags;
+    u32 bg_reserved[2];
+    u16 bg_itable_unused; // Unused inodes count
+    u16 bg_checksum;      // crc16(volume_uuid+group_num+group_desc)
+};
+
+// Structure of an inode on the disk
+struct Inode {
+    u16 mode;        // File mode
+    u16 owner_uid;   // Low 16 bits of Owner Uid
+    u32 file_size;   // Size in bytes
+    u32 access_time; // Access time
+    u32 create_time; // Inode change time
+    u32 modify_time; // Modification time
+    u32 delete_time; // Deletion Time
+    u16 group_id;    // Low 16 bits of Group Id
+    u16 num_links;   // Links count
+    u32 num_blocks;  // Blocks count
+    u32 file_flags;  // File flags
+
+    // This is Linux dependent
+    u32 linux_version;
+
+    u32 block_ptrs[Ext2::num_blocks]; // Pointers to blocks
+    u32 file_generation;              // File version (for NFS)
+    u32 file_acl;                     // File ACL
+    u32 directory_acl;                // Directory ACL
+    u32 fragment_addr;                // Fragment address
+
+    // These fields below are also Linux dependent.
+    u16 num_blocks_high;
+    u16 file_acl_high;
+    u16 owner_uid_high;
+    u16 group_id_high;
+    u32 reserved2;
+};
+
+// Revision levels
+enum class Revision : u32 {
+    Original, // The good old (original) format
+    Dynamic,  // V2 format w/ dynamic inode sizes
```

Comment:
I assume those revisions have actual set numbers that they should be using? Is it worth it to explicitly specify them here? Same for `enum FileType`.

Reply:
If the values in an enum are defined to start at 0 I'm pretty sure we can leave the values themselves out